### DISCUSSION TITLE
Javascript injection problem

### DISCUSSION THREAD

**Discussion by ZaakaJarimbo** [Alex-D/Trumbowyg#1397] [General]
I currenly have the following JS code to insert text into a <textarea> however when I use Trumbowyg to enhance the textarea this function stops working, How would I change the existing code to work with Trumbowyg?

`
function pad(num, size) {
    var s = "0" + num;
    return s.substr(s.length-size);
}    
      
    function settvref() {
    var seriesname = document.getElementById("series").value;
    var seasonnum = document.getElementById("season").value;
    var episodenum = document.getElementById("episode").value;
    var episodetitle = document.getElementById("episodetitle").value;
    document.getElementById("tvref").value + '<li>'+seriesname+' S'+pad(seasonnum,2)+'E'+pad(episodenum,2)+' ('+episodetitle+')</li>';
}
 `

**Comment by dev-alihasan** [comment]
you need to use Trumbowyg's API to get the content of the enhanced textarea and then append your new content to it

`function pad(num, size) {
  var s = "0" + num;
  return s.substr(s.length-size);
}

function settvref() {
  var seriesname = document.getElementById("series").value;
  var seasonnum = document.getElementById("season").value;
  var episodenum = document.getElementById("episode").value;
  var episodetitle = document.getElementById("episodetitle").value;

  // Get the current content of the Trumbowyg-enhanced textarea
  var currentContent = $('#tvref').trumbowyg('html');

  // Append your new content to the current content
  var newContent = currentContent + '<li>'+seriesname+' S'+pad(seasonnum,2)+'E'+pad(episodenum,2)+' ('+episodetitle+')</li>';

  // Set the updated content back into the Trumbowyg-enhanced textarea
  $('#tvref').trumbowyg('html', newContent);
}
`

**Reply by ZaakaJarimbo** [reply]
Thankyou, I will give it a try :)

**Comment by ZaakaJarimbo** [comment]
Working a treat¿, Thank you
